src/Microsoft.AspNet.OData.Shared/IDeltaSetItem.cs

```diff
@@ -0,0 +1,38 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="IDeltaSetItem.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using Microsoft.AspNet.OData.Builder;
+using Microsoft.OData.UriParser;
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// Basic interface for representing a delta item like delta, deleted entity, etc.
+    /// </summary>
+    public interface IDeltaSetItem
+    {
+        /// <summary>
+        /// Gets the kind of object within the delta payload.
+        /// </summary>
+        EdmDeltaEntityKind DeltaKind { get; }
+
+        /// <summary>
+        /// Gets or sets the annotation container to hold transient instance annotations.
+        /// </summary>
+        IODataInstanceAnnotationContainer TransientInstanceAnnotationContainer { get; set; }
+
+        /// <summary>
+        /// Gets or sets the container to hold ODataId
+        /// </summary>
+        IODataIdContainer ODataIdContainer { get; set; }
```

Comment:
>IODataIdContainer [](http://example.com/codeflow?start=8&length=17)

why don't use "string ODataId" directly? I mean why do we need a wrapper to wrap a string property?

Reply:
The interface was introduced to hold 'ODataId - @odata.id'. When doing a deep insert for instance, users are required to include the interface in their POCO classes if they want to use `@odata.id` to reference existing resources. The same was also added to `Delta<TStructuralType>`

Reply:
```c#  public class Press
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Email { get; set; }
        public Category Category { get; set; }
        public ODataIdContainer Container { get; set; }
    }
```

Reply:
What would be the problem if the string was passed directly?

Reply:
@habbes  then how will we tell that the string represents an `odata.id`